diff --git a/QtCompressMimeDatabase.cmake.orig b/QtCompressMimeDatabase.cmake
index 718d330..d6fc4be 100644
--- a/QtCompressMimeDatabase.cmake.orig
+++ b/QtCompressMimeDatabase.cmake
@@ -134,6 +134,9 @@ endif()
 
 string(REGEX MATCHALL "([a-f0-9][a-f0-9])" qmime_db_hex "${qmime_db_data}")
 
+list(REMOVE_AT qmime_db_hex 4 5 6 7)
+list(INSERT qmime_db_hex 4 "00" "00" "00" "00")
+
 list(TRANSFORM qmime_db_hex PREPEND "0x")
 math(EXPR qmime_db_data_size "${qmime_db_data_size} - 1")
 foreach(index RANGE 0 ${qmime_db_data_size} 12)
